test/e2e/plugin_lifecycle_test.go

```diff
@@ -0,0 +1,155 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	greenhouseapis "github.com/cloudoperators/greenhouse/pkg/apis"
+	greenhousev1alpha1 "github.com/cloudoperators/greenhouse/pkg/apis/greenhouse/v1alpha1"
+	"github.com/cloudoperators/greenhouse/pkg/test"
+	"github.com/cloudoperators/greenhouse/test/e2e/fixtures"
+)
+
+var _ = Describe("PluginLifecycle", Ordered, func() {
+	It("should deploy the plugin", func() {
+
+		const clusterName = "test-cluster-a"
+		setup := test.NewTestSetup(test.Ctx, test.K8sClient, "teamrbac")
+		secret := setup.CreateSecret(test.Ctx, clusterName,
+			test.WithSecretType(greenhouseapis.SecretTypeKubeConfig),
+			test.WithSecretData(map[string][]byte{greenhouseapis.KubeConfigKey: remoteKubeConfig}))
+		cluster := &greenhousev1alpha1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      clusterName,
+				Namespace: setup.Namespace(),
+			},
+		}
+		test.EventuallyCreated(test.Ctx, test.K8sClient, cluster)
+
+		testPluginDefinition := fixtures.NginxPluginDefinition
+		testPluginDefinition.ObjectMeta.Namespace = setup.Namespace() //namespace override
+
+		testPlugin := &greenhousev1alpha1.Plugin{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Plugin",
+				APIVersion: greenhousev1alpha1.GroupVersion.String(),
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-nginx-plugin",
+				Namespace: setup.Namespace(),
+			},
+			Spec: greenhousev1alpha1.PluginSpec{
+				PluginDefinition: "nginx-18.1.7",
+				ReleaseNamespace: setup.Namespace(),
+				ClusterName:      secret.Name,
+				OptionValues: []greenhousev1alpha1.PluginOptionValue{
+					{
+						Name:  "replicaCount",
+						Value: &apiextensionsv1.JSON{Raw: []byte("1")},
+					},
+				},
+			},
+			Status: greenhousev1alpha1.PluginStatus{},
+		}
+
+		pluginDefinitionList := &greenhousev1alpha1.PluginDefinitionList{}
+		pluginList := &greenhousev1alpha1.PluginList{}
+		deploymentList := &appsv1.DeploymentList{}
+		ctx := test.Ctx
+
+		//Creating plugin definition
+		err := test.K8sClient.Create(ctx, testPluginDefinition)
+		Expect(err).NotTo(HaveOccurred())
+		err = test.K8sClient.List(ctx, pluginDefinitionList)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(pluginDefinitionList.Items)).To(BeEquivalentTo(1))
+
+		//Creating plugin
+		err = test.K8sClient.Create(ctx, testPlugin)
+		Expect(err).NotTo(HaveOccurred())
+		err = test.K8sClient.List(ctx, pluginList)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(pluginList.Items)).To(BeEquivalentTo(1))
```

Comment:
Could you use the gomega's `Eventually(..)` to check the `testPlugin` eventually has it's status set to ready? If that does not happen, the test can already fail.

Reply:
Done